File: Problems/Leetcode/3146. Permutation Difference between Two Strings/main.ts
function sum(){
    let total = 0;
    return (s = 0, t = 0) => {
        return total += Math.abs(s - t);
    }
}
function findPermutationDifference(s: string, t: string): number {
    const add = sum();
    for(let i = 0; i < s.length; i++){
        add(i, t.indexOf(s[i]));
    }
    return add();
};  
console.log(findPermutationDifference("abc", "bac"))

File: Problems/Leetcode/3146. Permutation Difference between Two Strings/main.cpp
#include<string>
#include<iostream>
#include<cmath>
using namespace std;
class Solution {
private:
    int total = 0;
    int add(int s = 0, int t = 0){
        return total += fabs(s - t);
    }
public:
    int findPermutationDifference(string s, string t) {
        for(int i = 0; i < s.length(); i++){
            this->add(i, t.find(s[i]));
        }
        return add();
    }
};
int main(){
    Solution solution;
    const int result = solution.findPermutationDifference("abc", "bac");
    cout << result << endl;
    return 0;
}